File: xamarin/android/PropertyCross/AmbiguousLocationsAdapter.cs
using System;
using Android.Widget;
using PropertyFinder.Model;
using System.Collections.Generic;
using Android.Content;
using Android.Views;

namespace com.propertycross.xamarin.android
{
	public class AmbiguousLocationsAdapter : ArrayAdapter<Location>
	{
		private Context context;
		private IList<Location> items;

		public AmbiguousLocationsAdapter (Context c, IList<Location> items)
			: base(c, Android.Resource.Layout.SimpleListItem1, items)
		{
			this.context = c;
			this.items = items;
		}

		public override View GetView(int position, View convertView, ViewGroup parent)
		{
			View view = convertView;
			
			if(view == null)
			{
				LayoutInflater li = (LayoutInflater)context.GetSystemService(Context.LayoutInflaterService);
				view = li.Inflate(Android.Resource.Layout.SimpleListItem1, parent, false);
			}
			
			Location item = items[position];
			if (item != null) {
				TextView text = (TextView) view.FindViewById(Android.Resource.Id.Text1);
				text.Text = item.DisplayName;
			}

			return view;
		}
	}
}

File: native/android/PropertyCross/app/src/main/java/com/propertycross/android/views/AmbiguousLocationsAdapter.java
package com.propertycross.android.views;

import java.util.List;

import com.propertycross.android.model.Location;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

public class AmbiguousLocationsAdapter extends ArrayAdapter<Location> {

    private Context context;
    private List<Location> items;

    public AmbiguousLocationsAdapter(Context context, List<Location> items) {
        super(context, android.R.layout.simple_list_item_1, items);
        this.context = context;
        this.items = items;
        }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        View view = convertView;

        if(view == null) {
            LayoutInflater li = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
            view = li.inflate(android.R.layout.simple_list_item_1, parent, false);
            }

        Location item = items.get(position);
        if (item != null) {
            TextView text = (TextView) view.findViewById(android.R.id.text1);
            text.setText(item.getDisplayName());
        }

        return view;
    }

}
